mono/mini/exceptions-amd64.c

```diff
@@ -919,29 +919,45 @@ mono_arch_handle_altstack_exception (void *sigctx, MONO_SIG_HANDLER_INFO_TYPE *s
 }
 
 #ifndef DISABLE_JIT
+
+static
 GSList*
-mono_amd64_get_exception_trampolines (gboolean aot)
+mono_amd64_get_exception_trampoline (gboolean rethrow, gboolean corlib, gboolean llvm_abs, gboolean resume_unwind,
+				     const char *tramp_name, gboolean aot, gboolean reg, GSList* tramps)
 {
 	MonoTrampInfo *info;
-	GSList *tramps = NULL;
 
-	/* LLVM needs different throw trampolines */
-	get_throw_trampoline (&info, FALSE, TRUE, FALSE, FALSE, "llvm_throw_corlib_exception_trampoline", aot, FALSE);
-	tramps = g_slist_prepend (tramps, info);
+	(void)get_throw_trampoline (&info, rethrow, corlib, llvm_abs, resume_unwind, tramp_name, aot, FALSE);
 
-	get_throw_trampoline (&info, FALSE, TRUE, TRUE, FALSE, "llvm_throw_corlib_exception_abs_trampoline", aot, FALSE);
-	tramps = g_slist_prepend (tramps, info);
+	// Register the trampolines or build up a list.
+	if (reg) {
```

Comment:
Will anything bad happen if this is called for the same trampoline multiple times with reg set each time?

Reply:
I admit, I didn't really think about it. This change is meant to basically be a no-op, but it is preparation for another change, or rather whittling away at another change.

It is true though, this kinda increases/changes the exposure of something.
That previously was inlined elsewhere.

I am inlined not to worry about.

And more so, in the later change, we do check for repeated registration, of JIT icalls.
And they do occur. But I assert, in that other change, equivalency of the reptition.
And I worked through what repitition does occur.

I think currently, last registration wins, and that is ok.
As long as there is no race -- registration occurs before use.
Which at least amd64 tries to do.

Reply:
See here for the direction this is going and please accept this as a small intermediate step that doesn't really change anything, but makes later diffs smaller:

https://github.com/mono/mono/pull/14047/files#diff-eba0fd0052bf3ec9dd60f2f3e90f19e1R9150